pkg/collector/corechecks/servicediscovery/module/envs_test.go

```diff
@@ -103,3 +105,67 @@ func memfile(name string, b []byte) (int, error) {
 
 	return fd, nil
 }
+
+func TestEnvScanner(t *testing.T) {
+	curPid := os.Getpid()
+	proc, err := process.NewProcess(int32(curPid))
+	require.NoError(t, err)
+
+	envMapGood, err := getEnvs(proc)
+	require.NoError(t, err)
+
+	// extract environment variables using scanner
+	envMapCurr, err := getEnvsUsingScan(proc, false)
+	require.NoError(t, err)
+	require.Equal(t, envMapGood, envMapCurr)
+
+	// test extraction of target variables
+	expectedMap := map[string]string{
+		envVarDdService:               "service1",
+		envVarDdTags:                  "tag1,tag2,tag3",
+		envVarDdInjectionEnabled:      "true",
+		envVarDiscoveryEnabled:        "true",
+		envVarOtelServiceName:         "service2",
+		apm.EnvCoreClrEnableProfiling: "1",
+	}
+	var expectedEnvs []string
+	for k, v := range expectedMap {
+		expectedEnvs = append(expectedEnvs, fmt.Sprintf("%s=%s", k, v))
+	}
+
+	createEnvsMemfd(t, expectedEnvs)
+
+	envMapCurr, err = getEnvsUsingScan(proc, true)
+	require.NoError(t, err)
+	require.Equal(t, envMapCurr, expectedMap)
+}
+
+func BenchmarkGetEnvs(b *testing.B) {
+	b.ResetTimer()
+	b.ReportAllocs()
+	for i := 0; i < b.N; i++ {
+		proc, err := customNewProcess(int32(os.Getpid()))
```

Comment:
Maybe move the customNewProcess() call outside of the timed portion of the benchmark?